rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/Rename.rsc

```diff
@@ -479,12 +479,43 @@ private Cursor rascalGetCursor(TModel ws, Tree cursorT) {
     return cursor(kind, min(locsContainingCursor.l), cursorName);
 }
 
-private set[str] rascalNameToEquivalentNames(str name) =
-    {name, startsWith(name, "\\") ? name : "\\<name>"};
+private bool rascalContainsName(loc l, str n) {
```

Comment:
If __after profiling__ this function still takes up a lot of time, we could change it so that it returns a function that given a loc returns a boolean. And we only build the partial parse trees for the names once.